apps/precinct-scanner/src/AppRoot.tsx

```diff
@@ -384,6 +395,26 @@ const AppRoot = ({
     )
   }, [dispatchAppState])
 
+  // Handle hardware observer subscription
+  useEffect(() => {
+    const printerStatusSubscription = hardware.printers
+      .pipe(map((printers) => Array.from(printers)))
+      .subscribe(async (printers) => {
+        const newHasPrinterAttached = printers.some(
+          ({ connected }) => connected
+        )
+        dispatchAppState({
+          type: 'updateHardwareState',
+          hardwareState: {
+            hasPrinterAttached: newHasPrinterAttached,
+          },
+        })
+      })
+    return () => {
+      printerStatusSubscription.unsubscribe()
+    }
+  }, [])
```

Comment:
Consider abstracting this into a `usePrinter`? Could be useful for follow-up if not now.

Reply:
Yeah I considered doing this but its actually only also done in bmd (election-manager doesn't need to know if a printer is connected or not) and bmd does other bmd-specific things when the printer connects (like resets the ballot) so it didn't seem worth abstracting yet.